assets/js/components/Dashboard/NewPaMessagePage.tsx

```diff
@@ -38,15 +44,26 @@ enum AudioPreview {
 const NewPaMessagePage = () => {
   const now = moment();
 
+  const associatedAlertState = useAssociatedAlertContext();
```

Comment:
My hunch is this shouldn't be in the global context. This is the only page that will need this info and there aren't many layers in this workflow that require it. Should this instead be three additional pieces of local state? I know it requires some additional props on the `AssociateAlertPage` component, but that seems more appropriate than putting it in global state. What do you think?

Reply:
I agree it does seem more appropriate, but the issue I ran into was that you can't pass the `setState` callbacks via `navigate` because they're not serializable. Using context is just pattern that I was familiar with which let me achieve what I needed to do, but I'd certainly rather go with a pattern that make more sense rather than bloating the global state. Any suggestions on alternative approaches?

Reply:
Actually I guess I could just not use React Router here and just conditionally render either the associate alerts page or the new PA message page based off a new flag

Reply:
Ah, but that means the sidebar still shows. Not sure if that's technically an issue but it isn't explicitly part of the design

Reply:
The visible Sidebar was called out by Mary in my branch so it might also be a problem here. I was able to handle it like [this](https://github.com/mbta/screenplay/blob/7cd5ecc41c4cecb6e41c4e6542d9af65b4c4de46/assets/css/dashboard/new-pa-message/select-stations-page.scss#L3). Not super elegant but might be the best way considering it doesn't have a route to filter by.